public/app/features/panel/query_editor_row.ts

```diff
@@ -24,6 +24,22 @@ export class QueryRowCtrl {
       this.panelCtrl.getCollapsedText = this.queryCtrl.getCollapsedText.bind(this.queryCtrl);
     }
   }
+
+  $onChanges(): void {
```

Comment:
Can you explain what this is for? Is this a fix for the toggle to edit mode when the query mode is not Application Insights?

Reply:
Sorry, I was trying to fix the fact that the little penil icon doesn't work until you switch away and back. But I don't think it quite worked (there is something else caching the wrong value higher up in the chain I couldn't find). I'll revert this.

There is a bug that when you first switch to "Application Insights" in he dropdown, you can't go into raw query mode, because the fact that "Azure Monitor" doesn't have a text edit mode is cached already, and it was _really_ getting in the way of my validation. I initially thought I'd introduced it, but I was incorrect.